cumulusci/salesforce_api/retrieve_profile_api.py

```diff
@@ -0,0 +1,302 @@
+
+        # Process the results
+        self.permissionable_entities = {}
+        self.permissionable_entities.update(
+            self._process_setupEntityAccess_results(result[SETUP_ENTITY_QUERY_NAME])
+        )
+        self.permissionable_entities.update(
+            self._process_sObject_results(result[SOBJECT_QUERY_NAME])
+        )
+        self.permissionable_entities.update(
+            self._process_customTab_results(result[CUSTOM_TAB_QUERY_NAME])
+        )
+
+        # Logs
+        self.logger.info("[Done]\n")
+
+        # Retrieve profiles with their flows
+        self.profile_flow = self._match_profile_to_flows(profiles)
+
+        return self.permissionable_entities, self.profile_flow
+
+    def _process_setupEntityAccess_results(self, result_list: List[dict]):
+        setupEntityAccess_dict = {
+            entity_type: [] for entity_type in SETUP_ENTITY_TYPES.keys()
+        }
+
+        for data in result_list:
+            entity_type = data["SetupEntityType"]
+            entity_id = data["SetupEntityId"]
+            if entity_type in SETUP_ENTITY_TYPES.keys():
+                setupEntityAccess_dict[entity_type].append(entity_id)
+
+        queries = {}
+        for entity_type, query_values in SETUP_ENTITY_TYPES.items():
+            if query_values and len(setupEntityAccess_dict[entity_type]) != 0:
+                queries[entity_type] = self._build_query(
+                    query_values.columns,
+                    query_values.table_name,
+                    {query_values.id_field: setupEntityAccess_dict[entity_type]},
+                )
+
+        result = RunParallelQueries._run_queries_in_parallel(queries, self._run_query)
+
+        extracted_values = {}
+        for entity_type, data in SETUP_ENTITY_TYPES.items():
+            if entity_type in result and data:
+                extracted_values.setdefault(data.package_xml_name, [])
+                for item in result[entity_type]:
+                    if (
+                        "NamespacePrefix" in item
+                        and item["NamespacePrefix"] is not None
+                    ):
+                        extracted_values[data.package_xml_name].append(
+                            f'{item["NamespacePrefix"]}__{item[data.columns[0]]}'
+                        )
+                    else:
+                        extracted_values[data.package_xml_name].append(
+                            item[data.columns[0]]
+                        )
+
+        return extracted_values
+
+    def _process_sObject_results(self, result_list: List[dict]):
+        sObject_list = [data["SobjectType"] for data in result_list]
+        return {SOBJECT_TYPE: sObject_list}
+
+    def _process_customTab_results(self, result_list: List[dict]):
+        customTab_list = [data["Name"] for data in result_list]
+        return {CUSTOM_TAB_TYPE: customTab_list}
+
+    def _match_profile_to_flows(self, profiles: List[str]):
```

Comment:
Hey @aditya-balachander ,
can we move the code to _retrieve_permissionable_entities function and run query for flows along with other queries?